### PR TITLE
Trilinos 12.0.1 (update)

### PR DESCRIPTION
**Pull request by davydden**
pull after https://github.com/Homebrew/homebrew-science/pull/2504

### CODE REVIEW

File: trilinos.rb [hunk 1]
```diff
@@ -150,7 +156,7 @@ def install
     # we still need to provide METIS_INCLUDE_DIRS so that metis.h is picked up on Linuxbrew.
     if build.with? "metis"
       args << "-DTPL_ENABLE_METIS:BOOL=ON"
-      args << "-DMETIS_LIBRARIES=#{Formula["metis"].opt_lib}/libmetis.a"
+      args << "-DTPL_METIS_LIBRARIES=#{Formula["metis"].opt_lib}/libmetis.dylib"
```
Comment (dpo) [inline trilinos.rb]: Linux has no dylibs.

File: trilinos.rb [hunk 2]
```diff
@@ -170,7 +176,7 @@ def install
     if build.with? "parmetis"
       # Ensure CMake picks up METIS 5 and not METIS 4.
       args << "-DTPL_ENABLE_ParMETIS:BOOL=ON"
-      args << "-DParMETIS_LIBRARIES=#{Formula["parmetis"].opt_lib}/libparmetis.a;#{Formula["metis"].opt_lib}/libmetis.a"
+      args << "-DTPL_ParMETIS_LIBRARIES=#{Formula["parmetis"].opt_lib}/libparmetis.a;#{Formula["metis"].opt_lib}/libmetis.dylib"
```
Comment (dpo) [inline trilinos.rb]: Here you forgot to change libparmetis, but again Linux has no dylibs.

File: trilinos.rb [hunk 3]
```diff
@@ -226,30 +234,86 @@ def install
 
 __END__
 diff --git a/cmake/TPLs/FindTPLUMFPACK.cmake b/cmake/TPLs/FindTPLUMFPACK.cmake
-index e26fc7e..f0b7ab6 100644
+index 963eb71..998cd02 100644
```
Comment (dpo) [inline trilinos.rb]: Were the patches submitted upstream?

Comment (davydden) [inline trilinos.rb]: yes

File: trilinos.rb (hunk 1 above)
Comment (davydden) [inline trilinos.rb]: i'll test it later on linux to see if it's needed at all...

File: trilinos.rb (hunk 2 above)
Comment (davydden) [inline trilinos.rb]: i didn't forgot to change libparemtis, it's indeed `libparmetis.a` which is installed

**Comment by davydden** [comment]
tested on Linuxbrew, all good for me.

**Pull request closed without merging**